source/d3d8to9_device.cpp

```diff
@@ -1060,6 +978,7 @@ HRESULT STDMETHODCALLTYPE Direct3DDevice8::GetPaletteEntries(UINT PaletteNumber,
 }
 HRESULT STDMETHODCALLTYPE Direct3DDevice8::SetCurrentTexturePalette(UINT PaletteNumber)
 {
+	PaletteNumber = (PaletteNumber & 0xFFF) | (PC_NOCOLLAPSE << 12);
```

Comment:
Yes, I see your point.  That is suppose to be the palette index not a `PALETTEENTRY` structure.  I did a bit more research here and found some things that may shed light on this issue.

First of all it seems that 64bit Windows does not support `PALETTEENTRY`.  I found [this](https://msdn.microsoft.com/ja-jp/subscriptions/dd144877.aspx) documentation on a Microsoft site:

> Number of entries in the system palette. This index is valid only if the device driver sets the RC_PALETTE bit in the RASTERCAPS index and is available only if the driver is compatible with 16-bit Windows.

Notice the last line "is available only if the driver is compatible with 16-bit Windows".  As you know 64bit Windows is **not** compatible with 16-bit Windows.  I think this is trying to say that palettes are only supported if 16-bit Windows is supported.  I can verify this as all attempts to use the `PALETTEENTRY` on my 64bit system (even with GDI) has resulted in a crash.

Furthermore after looking closely at the `PC_NOCOLLAPSE` documentation I found [here](https://msdn.microsoft.com/en-us/library/cc250401.aspx) it says:

> Specifies that the color be placed in an unused entry in the system palette instead of being matched to an existing color in the system palette. If there are no unused entries in the system palette, the color is matched normally. Once this color is in the system palette, colors in other logical palettes can be matched to this color.

Notice it says "If there are no unused entries in the system palette, the color is matched normally."  It seems that the `PC_NOCOLLAPSE` will match the color normally if no system entry palettes exist, which is the case on my system.

I also read somewhere (though I cannot find it now) that the `PaletteNumber` used by `SetPaletteEntries` somehow corresponds to the `PALETTEENTRY`.

So as far as I can tell the reason `PC_NOCOLLAPSE` is working here is because Direct3D9 sees that there is no system palettes available and just matches this color normally.